Gibe.Umbraco.Blog/Composing/BlogCompositionComponent.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Gibe.Umbraco.Blog.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Models;
+using Umbraco.Core.PropertyEditors;
+using Umbraco.Core.Services;
+
+namespace Gibe.Umbraco.Blog.Composing
+{
+	public class BlogCompositionComponent : IComponent
+	{
+		private readonly IContentTypeService _contentTypeService;
+		private readonly IDataTypeService _dataTypeService;
+		private readonly IBlogSettings _blogSettings;
+		private readonly PropertyEditorCollection _propertyEditors;
+
+		private string _blogPostCompositionDocumentTypeAlias => $"blogPostComposition";
+
+		public BlogCompositionComponent(IContentTypeService contentTypeService,
+			IDataTypeService dataTypeService,
+			IBlogSettings blogSettings,
+			PropertyEditorCollection propertyEditors)
+		{
+			_contentTypeService = contentTypeService;
+			_dataTypeService = dataTypeService;
+			_blogSettings = blogSettings;
+			_propertyEditors = propertyEditors;
+		}
+
+		public void Initialize()
+		{
+			var existingBlogComposition = _contentTypeService.Get(_blogPostCompositionDocumentTypeAlias);
```

Comment:
How quick is this, I guess it's just a DB hit?

Reply:
Yeah, and it's only once on startup too. It's still not ideal, but we'd have to set a flag in some other way if we don't do this.